python/pyarrow/src/arrow/python/common.cc

```diff
@@ -131,6 +138,34 @@ class PythonErrorDetail : public StatusDetail {
   }
 
  protected:
+  Result<std::string> FormatImpl() const {
+    PyAcquireGIL lock;
+
+    // Use traceback.format_exception()
+    OwnedRef traceback_module;
+    RETURN_NOT_OK(internal::ImportModule("traceback", &traceback_module));
```

Comment:
It seems we are already retrieving the error indicator and saving the traceback in `PythonErrorDetail` class, but can be reading the code wrong.
Would it be possible to use `exc_traceback_.obj()` similar to `exc_type_.obj()` so there would be no need to use the traceback module?

Reply:
> It seems we are already retrieving the error indicator and saving the traceback in PythonErrorDetail class, but can be reading the code wrong.

Yes, the `traceback` module isn't being used to retrieve any errors. They are definitely already retrieved.

I am using `traceback.format_exception()` so that our error format is consistent with what Python prints normally, which should be easiest for users to understand. I could instead manually try to reproduce this format, but there's a lot of special logic in https://github.com/python/cpython/blob/3.12/Lib/traceback.py

Reply:
Ah, I see. Thank you for explaining!